### PR TITLE
Add unit tests for the gutenberg_render_block_core_post_title() function

### PR DESCRIPTION
**Pull request by akasunil**
## What?

Added unit test for Post title core block render function.

### CODE REVIEW

**Comment by akasunil** [comment]
@draganescu Can you please review this PR ?

**Comment by akasunil** [comment]
waiting for PR review

**Review by draganescu — DISMISSED** [review]
Tested the test and the test passes and it's a great addition. Thank you 👏🏻

**Comment by akasunil** [comment]
Thank you @draganescu. @Mamaduka can we merge this PR ?

File: phpunit/blocks/render-block-post-title-test.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	 *
+	 * @var int
+	 */
+	protected static $attributes;
+
+	/**
+	 * Setup method.
+	 */
+	public static function wpSetUpBeforeClass() {
+		self::$post = self::factory()->post->create_and_get( array( 'post_title' => 'Post title block Unit Test' ) );
```
Comment (spacedmonkey) [inline phpunit/blocks/render-block-post-title-test.php]:
```suggestion
	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
		self::$post = $factory->post->create_and_get( array( 'post_title' => 'Post title block Unit Test' ) );
```

Comment (akasunil) [reply phpunit/blocks/render-block-post-title-test.php]: Thank you for your review @spacedmonkey. I have updated the code according to your suggestion.

**Comment by akasunil** [comment]
waiting for PR review

File: phpunit/blocks/render-block-post-title-test.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Post Title rendering tests.
+ *
+ * @package WordPress
+ * @subpackage Blocks
```
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-title-test.php]:
```suggestion
 * @subpackage Blocks
 * @covers ::gutenberg_render_block_core_post_title
```

File: phpunit/blocks/render-block-post-title-test.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	 */
+	public function test_gutenberg_render_block_core_post_title() {
+
+		$block           = new stdClass();
+		$GLOBALS['post'] = self::$post;
+		$block->context  = array( 'postId' => self::$post->ID );
+
+		$rendered = gutenberg_render_block_core_post_title( self::$attributes, '', $block );
+		$this->assertNotNull( $rendered );
+		$this->assertStringContainsString( 'Post title block Unit Test', $rendered );
```
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-title-test.php]: When a test method contains more than 1 assertion, a failure message should be added to the assertion method.
Please see https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions

File: phpunit/blocks/render-block-post-title-test.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+ * @package WordPress
+ * @subpackage Blocks
+ */
+
+/**
+ * Tests for the  Post Title block.
+ *
+ * @group blocks
+ */
+class Tests_Blocks_Render_Post_Title extends WP_UnitTestCase {
```
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-title-test.php]:
```suggestion
class Tests_Blocks_GutenbergRenderBlockCorePostTitle {
```

```
Test class names should reflect the filepath, with underscores replacing directory separators and TitleCase replacing camelCase. Thus, the class Tests_Comment_GetCommentClass(), which contains tests for the get_comment_class() function, is located in tests/comment/getCommentClass.php.
```
Please see https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes

**Review by anton-vlasenko — COMMENTED** [review]
Please find my notes below.
Thanks.

**Comment by akasunil** [comment]
@anton-vlasenko Feedback addressed. Can you please review again.

File: phpunit/blocks/render-block-post-title-test.php [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+ * @package WordPress
+ * @subpackage Blocks
+ * @covers ::gutenberg_render_block_core_post_title
+ */
+
+/**
+ * Tests for the  Post Title block.
+ *
+ * @group blocks
+ */
```
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-title-test.php]: I'd make this a single docblock for simplicity.
```suggestion
<?php
/**
 * Tests for the gutenberg_render_block_core_post_title() function.
 *
 * @package WordPress
 * @subpackage Blocks
 *
 * @group blocks
 *
 * @covers ::gutenberg_render_block_core_post_title
 */
```

File: phpunit/blocks/render-block-post-title-test.php [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+		$block->context = array( 'postId' => 0 );
+		$rendered       = gutenberg_render_block_core_post_title( self::$attributes, '', $block );
+		$this->assertEmpty( $rendered );
+	}
+
+	/**
+	 * Test gutenberg_render_block_core_post_title() method.
+	 *
+	 * @covers ::gutenberg_render_block_core_post_title
+	 */
```
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-title-test.php]: You don't have to use the @covers tag for individual test methods if you have it defined in the header.
```suggestion
	/**
	 * Test gutenberg_render_block_core_post_title() method.
	 */
```

File: phpunit/blocks/render-block-post-title-test.php [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+ * @subpackage Blocks
+ * @covers ::gutenberg_render_block_core_post_title
+ */
+
+/**
+ * Tests for the  Post Title block.
+ *
+ * @group blocks
+ */
+class Tests_Blocks_GutenbergRenderBlockCorePostTitle extends WP_UnitTestCase {
```
Comment (anton-vlasenko) [inline phpunit/blocks/render-block-post-title-test.php]:
```suggestion
class Tests_Blocks_RenderBlockCorePostTitle extends WP_UnitTestCase {
```

**Review by anton-vlasenko — COMMENTED** [review]
https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes:
> Test files should be named using camelCase.

Could the test file be renamed to `renderBlockCorePostTitle.php`?

**Comment by akasunil** [comment]
> Could the test file be renamed to `renderBlockCorePostTitle.php`?

Is this the new standards that we are going to follow from now on for test file name? because all other files are named in kebab-case.

File: phpunit/blocks/render-block-post-title-test.php (hunk 7 above)
Comment (akasunil) [reply phpunit/blocks/render-block-post-title-test.php]: As per the standards you mentioned above, shouldn't we use the function name which is `gutenberg_render_block_core_post_title()`  in title case?

Comment (anton-vlasenko) [reply phpunit/blocks/render-block-post-title-test.php]: I don't have a strong preference on this one.
It's up to you.
If the code is going to be merged into WordPress Core at some point, I'd suggest aligning it with WordPress Core standards to ease the backporting efforts (and increase the chances of acceptance).

**Comment by anton-vlasenko** [comment]
> > Could the test file be renamed to `renderBlockCorePostTitle.php`?
> 
> Is this the new standards that we are going to follow from now on for test file name? because all other files are named in kebab-case.

No, this is not a new standard, but I agree, it might seem new. There are (mostly) old tests that don't follow this naming convention, and the newer ones that adhere to it, e.g.: https://github.com/WordPress/wordpress-develop/tree/trunk/tests/phpunit/tests/option

**Comment by akasunil** [comment]
@anton-vlasenko  To follow the standard for file name of unit tests, It required to update [phpunit config](https://github.com/WordPress/gutenberg/blob/trunk/phpunit.xml.dist#L14-L17) file first.

**Comment by akasunil** [comment]
I have updated phpunit config temporary until we change all filenames to camel case as per wordpress core standard. And Addressed all feedback provided by reviewer.

**Comment by akasunil** [comment]
@anton-vlasenko can you please review this PR?

File: phpunit/blocks/renderBlockCorePostTitle.php [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+ * @subpackage Blocks
+ *
+ * @covers ::gutenberg_render_block_core_post_title
+ * @group blocks
+ */
+class Tests_Blocks_RenderBlockCorePostTitle extends WP_UnitTestCase {
+	/**
+	 * Post object.
+	 *
+	 * @var array
```
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostTitle.php:15]:
```suggestion
	 * @var WP_Post
```

File: phpunit/blocks/renderBlockCorePostTitle.php [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+	 */
+	protected static $post;
+
+	/**
+	 * Array of attributes.
+	 *
+	 * @var int
```
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostTitle.php:22]:
```suggestion
	 * @var array
```

File: phpunit/blocks/renderBlockCorePostTitle.php [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+	public function test_should_render_empty_string_when_title_is_empty() {
+
+		// call render method with block context.
+		$rendered = gutenberg_render_block_core_post_title( self::$attributes, '', self::$block );
+		$this->assertEmpty( $rendered );
+
+		self::$block->context = array( 'postId' => 0 );
+		$rendered             = gutenberg_render_block_core_post_title( self::$attributes, '', self::$block );
+		$this->assertEmpty( $rendered );
+	}
```
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostTitle.php]: I think a data provider could be utilized here.

```suggestion
	/**
	 * Test if the function returns an empty string when the post title is empty.
	 *
	 * @dataProvider data_should_render_empty_string_when_title_is_empty
	 */
	public function test_should_render_empty_string_when_title_is_empty( $context, $failure_message ) {
		if ( $context ) {
			self::$block->context = $context;
		}

		$rendered = gutenberg_render_block_core_post_title( self::$attributes, '', self::$block );
		$this->assertSame( '', $rendered, $failure_message );
	}

	/**
	 * Data provider.
	 *
	 * @return array
	 */
	public function data_should_render_empty_string_when_title_is_empty() {
		return array(
			'empty block context' => array( null, 'Failed asserting that rendering result is an empty string when block context is not set.' ),
			'incorrect post ID'   => array( array( 'postId' => 0 ), 'Failed asserting that rendering result is an empty string when post ID is incorrect.' ),
		);
	}
```

File: phpunit/blocks/renderBlockCorePostTitle.php [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+	/**
+	 * Test gutenberg_render_block_core_post_title() method.
```
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostTitle.php]: `gutenberg_render_block_core_post_title()` is not a method.
```suggestion
	 * Test if the function returns correct titles.
```

File: phpunit/blocks/renderBlockCorePostTitle.php (hunk 9 above)
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostTitle.php]:
```suggestion
	 * @var array
```

File: phpunit/blocks/renderBlockCorePostTitle.php (hunk 8 above)
Comment (anton-vlasenko) [inline phpunit/blocks/renderBlockCorePostTitle.php]:
```suggestion
	 * @var WP_Term
```

**Comment by akasunil** [comment]
@anton-vlasenko Can you please review this PR again?

**Review by anton-vlasenko — APPROVED** [review]
LGTM.

**Pull request merged**